p2p/gossip/protocol.go

```diff
@@ -138,18 +141,23 @@ func (prot *Protocol) Close() {
 	close(prot.shutdown)
 }
 
-// markMessage adds the calcHash to the old message queue so the message won't be processed in case received again
-func (prot *Protocol) markMessage(h hash) {
+// markMessage adds the calcHash to the old message queue so the message won't be processed in case received again.
+// Returns true if message was already processed before
+func (prot *Protocol) markMessage(h hash) bool {
 	prot.oldMessageMu.Lock()
-	prot.oldMessageQ[h] = struct{}{}
+	var ok bool
+	if _, ok = prot.oldMessageQ[h]; !ok {
+		prot.oldMessageQ[h] = struct{}{}
+	}
+	prot.Log.Debug("marking message as old, hash %v, is already old %v", h, ok)
```

Comment:
The message is confusing, you are not marking the message as old here.

Reply:
The "is already old" tell you whether it was already old